status-resolver/src/main/java/com/ackee/versioupdatehandler/model/DialogSettings.kt

```diff
@@ -0,0 +1,146 @@
+package com.ackee.versioupdatehandler.model
+
+import android.os.Parcel
+import android.os.Parcelable
+
+/**
+ * Visual settings of dialog
+ */
+class DialogSettings : Parcelable {
+
+    companion object {
+        val TAG = DialogSettings::class.java.name
+        val CREATOR: Parcelable.Creator<DialogSettings> = object : Parcelable.Creator<DialogSettings> {
+            override fun createFromParcel(parcelableValue: Parcel): DialogSettings? {
+                return DialogSettings(parcelableValue)
+            }
+
+            override fun newArray(size: Int): Array<DialogSettings?> {
+                return arrayOfNulls(size)
+            }
+        }
+    }
+
+    var packageName: String?
+    var title: String?
+    var titleRes: Int
+    var message: String?
+    var messageRes: Int
+    var positiveButton: String?
+    var positiveButtonRes: Int
+    var negativeButton: String?
+    var negativeButtonRes: Int
+
+    private constructor(builder: Builder) {
+        packageName = builder.packageName
+        title = builder.title
+        titleRes = builder.titleRes
+        message = builder.message
+        messageRes = builder.messageRes
+        negativeButton = builder.negativeButton
+        positiveButton = builder.positiveButton
+        positiveButtonRes = builder.positiveButtonRes
+        negativeButtonRes = builder.negativeButtonRes
+    }
+
+    protected constructor(parcelableValue: Parcel) {
+        packageName = parcelableValue.readString()
+        title = parcelableValue.readString()
+        message = parcelableValue.readString()
+        positiveButton = parcelableValue.readString()
+        negativeButton = parcelableValue.readString()
+        messageRes = parcelableValue.readInt()
+        titleRes = parcelableValue.readInt()
+        positiveButtonRes = parcelableValue.readInt()
+        negativeButtonRes = parcelableValue.readInt()
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    override fun writeToParcel(dest: Parcel, flags: Int) {
+        dest.writeString(packageName)
+        dest.writeString(title)
+        dest.writeString(message)
+        dest.writeString(positiveButton)
+        dest.writeString(negativeButton)
+        dest.writeInt(messageRes)
+        dest.writeInt(titleRes)
+        dest.writeInt(positiveButtonRes)
+        dest.writeInt(negativeButtonRes)
+    }
+
+    class Builder {
```

Comment:
remove the builder pls. We have kotlin now which can leverage default parameters. i dont think that we've used this dialog settings in any project at all

Reply:
71ab9eb15dbb56b1d722db15cc7c3d3f1daa120a